bundles/binding/org.openhab.binding.rwesmarthome/src/main/java/org/openhab/binding/rwesmarthome/internal/RWESmarthomeGenericBindingProvider.java

```diff
@@ -0,0 +1,217 @@
+		
+		logger.info("Adding item {} with {}", item.getName(), config.toString());
+		items.put(item.getName(), item);
+		
+		itemMapById.put(config.getDeviceId(), item);
+		itemMapByIdAndParam.put(config.getDeviceId(), config.getDeviceParam(), item);
+		
+		addBindingConfig(item, config);
+		this.context.setBindingChanged(true);
+	}
+	
+	
+	/**
+	 * This is a helper class holding binding specific configuration details
+	 * 
+	 * @author ollie-dev
+	 * @since 1.8.0
+	 */
+	public class RWESmarthomeBindingConfig implements BindingConfig {
+		private String deviceId = null;
+		private String deviceParam = null;
+		
+		/**
+		 * @return the device id
+		 */
+		public String getDeviceId() {
+			return deviceId;
+		}
+
+		/**
+		 * Sets the deviceId
+		 * 
+		 * @param deviceId
+		 */
+		public void setDeviceId(String deviceId) {
+			this.deviceId = deviceId;
+		}		
+		
+		/**
+		 * @return device parameter
+		 */
+		public String getDeviceParam() {
+			return deviceParam;
+		}
+
+		/**
+		 * @param deviceParam
+		 */
+		public void setDeviceParam(String deviceParam) {
+			this.deviceParam = deviceParam;
+		}
+
+		@Override
+		public String toString() {
+			String configString = "RWESmarthomeBindingConfig [deviceId=" + deviceId + ", param=" + deviceParam + "]";
+			return configString;
+		}
+
+	}
+
+
+	/* (non-Javadoc)
+	 * @see org.openhab.binding.rwesmarthome.RWESmarthomeBindingProvider#getItem(java.lang.String)
+	 */
+	@Override
+	public Item getItem(String itemName) {
+		return items.get(itemName);
+	}
+
+	/* (non-Javadoc)
+	 * @see org.openhab.binding.rwesmarthome.RWESmarthomeBindingProvider#getItemById(java.lang.String)
+	 */
+	@Override
+	public Item getItemById(String id) {
+		return (Item) itemMapById.get(id);
+	}
+
+	/* (non-Javadoc)
+	 * @see org.openhab.binding.rwesmarthome.RWESmarthomeBindingProvider#getItemByIdAndParam(java.lang.String, java.lang.String)
+	 */
+	@Override
+	public Item getItemByIdAndParam(String id, String param) {
+		return (Item) itemMapByIdAndParam.get(id, param);
+	}
+
+	/* (non-Javadoc)
+	 * @see org.openhab.core.binding.BindingChangeListener#bindingChanged(org.openhab.core.binding.BindingProvider, java.lang.String)
+	 */
+	@Override
+	public void bindingChanged(BindingProvider provider, String itemName) {
+		logger.debug("BINDING CHANGED! item='{}', provider='{}'", itemName, provider);
+		context.setBindingChanged(true);
+	}
+
+	/* (non-Javadoc)
+	 * @see org.openhab.core.binding.BindingChangeListener#allBindingsChanged(org.openhab.core.binding.BindingProvider)
+	 */
+	@Override
+	public void allBindingsChanged(BindingProvider provider) {
+		logger.debug("ALL BINDINGS CHANGED! provider='{}'", provider);
+		context.setBindingChanged(true);
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.openhab.binding.rwesmarthome.RWESmarthomeBindingProvider#getBindingFor(java.lang.String)
+	 */
+	public RWESmarthomeBindingConfig getBindingFor(String itemName) {
```

Comment:
rename to getBindingConfigFor() since a configuration is returned and not the Binding